### ISSUE TITLE
Bad Certificate when using HTTPS

### ISSUE THREAD

**Issue by Couto** [thedevs-network/kutt#250]
After setting up my own domain on your instance, I get a bad certificate message since the domains don't match, whenever I try to visit my homepage.

Is this something temporary while you use Let's Encrypt to generate the certificate for my custom domain, or is this a problem?

![Screen Shot 2019-12-24 at 09 57 52](https://user-images.githubusercontent.com/165314/71407838-14f55a00-2634-11ea-8a30-c119d2eb1eb1.png)

![Screen Shot 2019-12-24 at 09 58 08](https://user-images.githubusercontent.com/165314/71407831-0dce4c00-2634-11ea-8b93-2fc254a19c46.png)

**Comment by Couto** [comment]
After some tests, this happens with any link when visiting with `https`. I thought the service would always redirect http:// to https:// but apparently not.

So any URL like: https://customdoma.in/something will trigger this error.

**Comment by poeti8** [comment]
Yes it currently doesn't handle HTTPS, but it's on my list to add it soon.

**Comment by martijnluyckx** [comment]
Do you have an estimate of when this will be implemented?

**Comment by poeti8** [comment]
I finished refactoring and I'm adding features one by one now, could take couple months. Until then, you can email me (support@kutt.it) and I can add cert for domains manually.

**Comment by lapphan** [comment]
It looks like HTTPS is working properly on custom domain associated with an [kutt.it](https://kutt.it) account now.

**Comment by lapphan** [comment]
It looks like HTTPS is working properly on custom domain associated with an [kutt.it](https://kutt.it) account now.
We can also close https://github.com/thedevs-network/kutt/issues/18.

**Comment by w4rner** [comment]
Another pre-requisite to use the product is that it supports https. Google is slamming non-https.

**Comment by lapphan** [comment]
My custom domain that added to kutt.it website has a SSL certificate from its DNS. If you can not support custom domains HTTPS on kutt.it, at least leave them untouched. What the hell is "get.forkful.app" 's certificate associated with my custom domain?

![image](https://user-images.githubusercontent.com/11986206/95005644-787dd080-0625-11eb-838d-e023997e9d4b.png)

**Comment by k-allika** [comment]
> I finished refactoring and I'm adding features one by one now, could take couple months. Until then, you can email me ([support@kutt.it](mailto:support@kutt.it)) and I can add cert for domains manually.

Sounds great! Looking forward to it.

In the meanwhile, may I request HTTPS support for my domain. I emailed you the information.

**Comment by w4rner** [comment]
what if you get free SSL in Cloudflare, can it work with https automatically?

**Comment by murka** [comment]
> what if you get free SSL in Cloudflare, can it work with https automatically?

@w4rner, yep, if you enable proxy, it will work.

![cloudflare](https://screenshots.niina.xyz/chrome_B3WsdByh1q.png)
![digicert](https://screenshots.niina.xyz/chrome_jN8G5hSOpX.png)

**Comment by trgwii** [comment]
Inactivity
